src/main/host/syscall/handler/mod.rs

```diff
@@ -0,0 +1,85 @@
+use crate::cshadow as c;
+use crate::host::context::{ThreadContext, ThreadContextObjs};
+use crate::host::syscall_types::SysCallArgs;
+use crate::host::syscall_types::{SyscallError, SyscallResult};
```

Comment:
What's the reason for making a new `handler` module? What files/types will go in the `syscall` module and not the `syscall::handler` module?

Also I think the existing functions in the syscall module (`src/main/host/syscall/mod.rs`) should also move into this `handler` module since they're helpers specifically for the handler functions.

Reply:
> What's the reason for making a new handler module? What files/types will go in the syscall module and not the syscall::handler module?

I was thinking we would place the syscall-related types and functions that are not specific to the handler itself in `src/main/host/syscall` and then isolate the handlers `impl`s in `src/main/host/syscall/handler`. So, for example, `syscall_condition`, `syscall_types`, and maybe `kernel_types` would go in `src/main/host/syscall`.

> Also I think the existing functions in the syscall module (src/main/host/syscall/mod.rs) should also move into this handler module since they're helpers specifically for the handler functions.

OK, thanks! I'm also planning to continue iterating on organization etc. as I move more handlers to Rust but this PR was already getting pretty big.